tests/baselines/reference/inlinedAliasAssignableToConstraintSameAsAlias.types

```diff
@@ -31,7 +31,7 @@ class A {
 >z : A[]
 
   whereRelated< // Works // Type is same as A1, but is not assignable to type A
->whereRelated : <RF extends RelationFields = RelationFields, N extends "x" | "y" | "z" = "x" | "y" | "z", A1 extends A = RF[N] extends A[] ? RF[N][0] : never, A2 extends A = ShouldA<RF, N>>() => number

```

Comment:
`ShouldA<RF, N>` is identical to `RF[N] extends A[] ? RF[N][0] : never`, and we make the second before the aliased version, so we cache a version with no alias symbol. Previously we'd make two different types because of the substitution in the `true` branch. :(